{
  "cve": "CVE-2022-35997",
  "mitre": {
    "cpes": [],
    "created": "2022-09-16T22:15:18+00:00",
    "description": "TensorFlow is an open source platform for machine learning. If `tf.sparse.cross` receives an input `separator` that is not a scalar, it gives a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 83dcb4dbfa094e33db084e97c4d0531a559e0ebf. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.",
    "metrics": {
      "cvssV2_0": {},
      "cvssV3_0": {},
      "cvssV3_1": {
        "score": 5.9,
        "vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H"
      },
      "cvssV4_0": {}
    },
    "mitre_repo_path": "cves/2022/35xxx/CVE-2022-35997.json",
    "references": [
      "https://github.com/tensorflow/tensorflow/commit/83dcb4dbfa094e33db084e97c4d0531a559e0ebf",
      "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-p7hr-f446-x6qf"
    ],
    "title": "`CHECK` fail in `tf.sparse.cross` in TensorFlow",
    "updated": "2024-08-03T09:51:59.513000+00:00",
    "vendors": [],
    "weaknesses": [
      "CWE-617"
    ]
  },
  "nvd": {
    "cpes": [
      "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.10:rc0:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.10:rc1:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.10:rc2:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.10:rc3:*:*:*:*:*:*"
    ],
    "created": "2022-09-16T23:15:10.467000+00:00",
    "description": "TensorFlow is an open source platform for machine learning. If `tf.sparse.cross` receives an input `separator` that is not a scalar, it gives a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 83dcb4dbfa094e33db084e97c4d0531a559e0ebf. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.",
    "metrics": {
      "cvssV2_0": {},
      "cvssV3_0": {},
      "cvssV3_1": {
        "score": 7.5,
        "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
      },
      "cvssV4_0": {}
    },
    "nvd_repo_path": "2022/CVE-2022-35997.json",
    "references": [
      "https://github.com/tensorflow/tensorflow/commit/83dcb4dbfa094e33db084e97c4d0531a559e0ebf",
      "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-p7hr-f446-x6qf"
    ],
    "title": null,
    "updated": "2022-09-20T14:49:39.707000+00:00",
    "vendors": [
      "google",
      "google$PRODUCT$tensorflow"
    ],
    "weaknesses": [
      "CWE-617"
    ]
  },
  "opencve": {
    "changes": [],
    "cpes": {
      "data": [
        "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*",
        "cpe:2.3:a:google:tensorflow:2.10:rc0:*:*:*:*:*:*",
        "cpe:2.3:a:google:tensorflow:2.10:rc1:*:*:*:*:*:*",
        "cpe:2.3:a:google:tensorflow:2.10:rc2:*:*:*:*:*:*",
        "cpe:2.3:a:google:tensorflow:2.10:rc3:*:*:*:*:*:*"
      ],
      "providers": [
        "nvd"
      ]
    },
    "created": {
      "data": "2022-09-16T22:15:18+00:00",
      "provider": "mitre"
    },
    "description": {
      "data": "TensorFlow is an open source platform for machine learning. If `tf.sparse.cross` receives an input `separator` that is not a scalar, it gives a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 83dcb4dbfa094e33db084e97c4d0531a559e0ebf. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.",
      "provider": "mitre"
    },
    "metrics": {
      "cvssV2_0": {
        "data": {},
        "provider": null
      },
      "cvssV3_0": {
        "data": {},
        "provider": null
      },
      "cvssV3_1": {
        "data": {
          "score": 5.9,
          "vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H"
        },
        "provider": "mitre"
      },
      "cvssV4_0": {
        "data": {},
        "provider": null
      },
      "kev": {
        "data": {},
        "provider": null
      },
      "ssvc": {
        "data": {},
        "provider": null
      },
      "threat_severity": {
        "data": null,
        "provider": null
      }
    },
    "references": {
      "data": [
        "https://github.com/tensorflow/tensorflow/commit/83dcb4dbfa094e33db084e97c4d0531a559e0ebf",
        "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-p7hr-f446-x6qf"
      ],
      "providers": [
        "mitre",
        "nvd"
      ]
    },
    "title": {
      "data": "`CHECK` fail in `tf.sparse.cross` in TensorFlow",
      "provider": "mitre"
    },
    "updated": {
      "data": "2024-08-03T09:51:59.513000+00:00",
      "provider": "mitre"
    },
    "vendors": {
      "data": [
        "google",
        "google$PRODUCT$tensorflow"
      ],
      "providers": [
        "nvd"
      ]
    },
    "weaknesses": {
      "data": [
        "CWE-617"
      ],
      "providers": [
        "mitre",
        "nvd"
      ]
    }
  }
}